[DOC] Improve usability of upgrade docs (#1528)
strimzi/strimzi-kafka-operator: documentation/book/proc-upgrading-brokers-newer-kafka.adoc

//
// assembly-upgrading-kafka-versions.adoc

[id='proc-upgrading-brokers-newer-kafka-{context}']

> = Upgrading Kafka brokers and client applications

Reviewer: I feel the more descriptive title was helpful. Please consider changing it to:

Upgrading Kafka brokers and client applications to a new Kafka version

Author: Have changed, but I get uncomfortable with long headings. :-)

> [id='proc-upgrading-brokers-newer-kafka-0-9-0-to-0-10-0-{context}']

Reviewer 2: I think we changed this largely to a general chapter. We should probably remove the reference to specific versions.

Author: Yes. Updated.

----
+
[source,shell,subs=+quotes]
----
oc get po -w
----
+
====
If the current and new versions of Kafka have different interbroker protocol versions, check the Cluster Operator logs for an `INFO` level message:

[source,shell,subs=+quotes]
----
Reconciliation #_<num>_(watch) Kafka(_<namespace>_/_<name>_): Kafka version upgrade from _<from-version>_ to _<to-version>_, phase 2 of 2 completed
----
Alternatively, if the current and new versions of Kafka have the same interbroker protocol version, check for:

[source,shell,subs=+quotes]
----
Reconciliation #_<num>_(watch) Kafka(_<namespace>_/_<name>_): Kafka version upgrade from _<from-version>_ to _<to-version>_, phase 1 of 1 completed
----
====
+
The updates:
+
* Ensure each pod is using the broker binaries for the new version
* Configures the brokers to send messages using the interbroker protocol of the new version
+
NOTE: Clients are still using the old version, so brokers will convert messages to the old version before sending them to the clients. To minimize this additional load, updates the clients as quickly as possible.

> . Upgrade all client applications (consumers) to use the new version of the client binaries.

Author: Excellent suggestion. Changed.

+
ifdef::Kubernetes[]
On {KubernetesName}, use:
+
[source,shell,subs=+quotes]
----
kubectl logs -f _<cluster-operator-pod-name>_ | grep -E "Kafka version upgrade from [0-9.]+ to [0-9.]+, phase ([0-9]+) of \1 completed"
----
+
[source,shell,subs=+quotes]
----
kubectl get po -w
----
+
endif::Kubernetes[]
On {OpenShiftName}, use:
+
[source,shell,subs=+quotes]
----
oc logs -f _<cluster-operator-pod-name>_ | grep -E "Kafka version upgrade from [0-9.]+ to [0-9.]+, phase ([0-9]+) of \1 completed"
----
+
[source,shell,subs=+quotes]
----
oc get po -w
----
+
====
If the current and new versions of Kafka have different interbroker protocol versions, check the Cluster Operator logs for an `INFO` level message:

[source,shell,subs=+quotes]
----
Reconciliation #_<num>_(watch) Kafka(_<namespace>_/_<name>_): Kafka version upgrade from _<from-version>_ to _<to-version>_, phase 2 of 2 completed
----
Alternatively, if the current and new versions of Kafka have the same interbroker protocol version, check for:

[source,shell,subs=+quotes]
----
Reconciliation #_<num>_(watch) Kafka(_<namespace>_/_<name>_): Kafka version upgrade from _<from-version>_ to _<to-version>_, phase 1 of 1 completed
----
====
+

> The updates:

Reviewer: ```suggestion
The rolling updates:
```

+

> * Ensure each pod is using the broker binaries for the new version

Reviewer: ```suggestion
* Ensure each pod is using the broker binaries for the new version of Kafka
```

> * Configures the brokers to send messages using the interbroker protocol of the new version

Reviewer: ```suggestion
* Configure the brokers to send messages using the interbroker protocol of the new version of Kafka
```

+
NOTE: Clients are still using the old version, so brokers will convert messages to the old version before sending them to the clients. To minimize this additional load, updates the clients as quickly as possible.

. Upgrade all client applications (consumers and producers) to use the new version of the client binaries.
+
See xref:con-strategies-for-upgrading-clients-str[]
+
WARNING: You cannot downgrade after completing this step. If you need to revert the update at this point, follow the procedure xref:proc-downgrading-brokers-older-kafka-{context}[].

+
If required, set the version property for Kafka Connect and Mirror Maker as the new version of Kafka:
+
.. For Kafka Connect, update `KafkaConnect.spec.version`

> .. For MIrror Maker, update `KafkaMirrorMaker.spec.version`

Reviewer: Typo: Mirror Maker

Author: Good spot! I did a double take to see it.
